_pytest/_code/code.py

```
@@ -110,7 +110,10 @@ def eval(self, code, **vars):
         """
         f_locals = self.f_locals.copy()
         f_locals.update(vars)
-        return eval(code, self.f_globals, f_locals)
+        try:
```

this change breaks the eval api hiding exceptions and changing the return value behaviour